api/envoy/api/v2/route/route.proto

```diff
@@ -921,6 +921,10 @@ message HeaderMatcher {
     //   "-1somestring"
     envoy.type.Int64Range range_match = 6;
   }
+
+  // If specified, header match will be performed based on if the header value is *not* the value
```

Comment:
nit: I would rephrase this to be a bit more generic as we may add future condition types and for it to better cover regex. For example: "If specified, the match result will be inverted before checking. E.g., does the regex not match, is the header value not present, etc.